### PR TITLE
Added CollectErrorMessagesModule

### PR DESCRIPTION
**Pull request by Didddy**

### CODE REVIEW

File: AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+namespace Ranorex.AutomationHelpers.Modules
+{
+    /// <summary>
+    /// Functionality: Collects all error messages from given test container.
+    /// Usage: Use module within a test container with enabled retry count
+    /// </summary>
+    [TestModule("E4E50BBC-20F8-461F-B39A-863B6F69DE89", ModuleType.UserCode, 1)]
+    public class CollectErrorMessagesModule : ITestModule
+    {
+        public static List<string> builder = new List<string>();
```
Comment (ahoisl) [inline AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs]: Should IMHO not be public

File: AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            {
+                GetErrorMessages((TestContainerActivity)container);
+            }
+        }
+
+        /// <summary>
+        /// Recursively collects all error messages from given test container
+        /// </summary>
+        /// <param name="container"></param>
+        private void GetErrorMessages(TestContainerActivity container)
```
Comment (ahoisl) [inline AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs]: Use interface ITestContainerActivity instead

File: AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                CollectErrorMessagesModule.retryIteration = 1;
+            }
+
+            //GetErrorMessages from TestContainer   
+            var container = TestReport.CurrentTestContainerActivity;
+
+            //Only collect error messages if the test run failed
+            if (container.Status == ActivityStatus.Failed)
+            {
+                GetErrorMessages((TestContainerActivity)container);
```
Comment (ahoisl) [inline AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs]: No need for cast if you use the interface in GetErrorMessages

File: AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+        /// <summary>
+        /// Recursively collects all error messages from given test container
+        /// </summary>
+        /// <param name="container"></param>
+        private void GetErrorMessages(TestContainerActivity container)
+        {
+            foreach (var containerChild in container.Children)
+            {
+                var containerChildType = containerChild.GetType().Name;
```
Comment (ahoisl) [inline AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs]: Uhhh, don't use type name (string), but the type directly, e.g.
if (containerChild is ITestContainerActivity)
{
// do something
}

File: AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+                //Check if TestContainerActivity
+                if (containerChildType == "TestContainerActivity")
+                {
+                    GetErrorMessages((TestContainerActivity)containerChild);
+                }
+
+                //Check if TestModuleActivity
+                if (containerChildType == "TestModuleActivity")
+                {
+                    var castedContainerChild = containerChild as TestModuleActivity;
```
Comment (ahoisl) [inline AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs]: Can you use the interface (ItestModuleActivity) instead?

File: AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+                    foreach (var testModuleChild in castedContainerChild.Children)
+                    {
+                        var testModuleChildType = testModuleChild.GetType().Name;
+
+                        //Check if testModuleChild is ReportItem
+                        if (testModuleChildType == "ReportItem")
+                        {
+                            var castedTestModuleChild = testModuleChild as ReportItem;
```
Comment (ahoisl) [inline AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs]: Can you use the interface (IReportItem) instead?

File: AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+                }
+
+                //Check if TestModuleActivity
+                if (containerChildType == "TestModuleActivity")
+                {
+                    var castedContainerChild = containerChild as TestModuleActivity;
+
+                    foreach (var testModuleChild in castedContainerChild.Children)
+                    {
+                        var testModuleChildType = testModuleChild.GetType().Name;
```
Comment (ahoisl) [inline AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs]: Don't use type names, see above...

**Comment by Didddy** [comment]
@ahoisl Thx for your review! I adapted your suggestions in #21

**Pull request merged**